### PR TITLE
Add create cluster pipeline

### PR DESCRIPTION
**Pull request by natifridman**
This PR is to initialize managed cluster creation pipeline.

PR includes:
- Tekton Pipeline
- Tekton PipelineRun
- Tekton task for copying files from one workspace folder to another

### CODE REVIEW

File: components/tekton/pipelineruns/create-cluster/create-cluster-pipelinerun.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  # yamllint disable rule:line-length
+  # Git-clone task runs as non-root user
+  # Generally, the default permissions for storage volumes are configured for the root user.
+  # To make the volumes accessible by the non-root user,
+  # you will need to either configure the permissions manually or set the fsGroup field
+  # under PodSecurityContext in your TaskRun or PipelineRun.
+  # yamllint enable rule:line-length
+  podTemplate:
+    securityContext:
+      fsGroup: 65532
```
Comment (nocturnalastro) [inline components/tekton/pipelineruns/create-cluster/create-cluster-pipelinerun.yaml:20]: Can you add a comment about what this means :)

Comment (natifridman) [reply components/tekton/pipelineruns/create-cluster/create-cluster-pipelinerun.yaml:20]: There is an explanation in the comments above. Would you like also to add a comment about the `fsGroup` number?

**Review by hhamid93 — APPROVED** [review]
other then some yaml lint warnings looks good.
A Readme.md would be helpful to exactly know how to use this pipeline specifically and what exactly will the pipeline deploy.

**Comment by natifridman** [comment]
> other then some yaml lint warnings looks good. A Readme.md would be helpful to exactly know how to use this pipeline specifically and what exactly will the pipeline deploy.

@hhamid93 Good point about documentation, I'll add some.
Yaml lint warnings are for files that are not in the scope of this PR, so I agree we should fix it but I think this should be a separate PR

File: components/tekton/pipelineruns/create-cluster/create-cluster-pipelinerun.yaml (hunk 1 above)
Comment (nocturnalastro) [reply components/tekton/pipelineruns/create-cluster/create-cluster-pipelinerun.yaml:20]: :facepalm: No its fine I didn't read the above comment.

**Review by nocturnalastro — APPROVED** [review]

**Comment by natifridman** [comment]
> @hhamid93 Good point about documentation, I'll add some. Yaml lint warnings are for files that are not in the scope of this PR, so I agree we should fix it but I think this should be a separate PR

Done

**Pull request merged**